rust/opendp/src/meas/geometric.rs

```diff
@@ -0,0 +1,55 @@
+use std::marker::PhantomData;
+use std::ops::{Sub, Add};
+
+
+use crate::core::{Function, Measurement, PrivacyRelation};
+use crate::dist::{L2Sensitivity, MaxDivergence};
+use crate::dom::AllDomain;
+use crate::error::Fallible;
+use crate::meas::{MakeMeasurement3, SampleBernoulli, SampleGeometric, SampleUniform};
+use crate::traits::DistanceCast;
+
+pub struct BaseSimpleGeometric<T> {
+    data: PhantomData<T>
+}
+
+// gaussian for scalar-valued query
+impl<T> MakeMeasurement3<AllDomain<T>, AllDomain<T>, L2Sensitivity<T>, MaxDivergence<f64>, f64, T, T> for BaseSimpleGeometric<T>
```

Comment:
Why is scale fixed to f64 here, when you've parameterized the others?

Reply:
`SampleBernoulli` has an interface that takes a probability as an f64. This f64 is re-exported in the `SampleGeometric` trait. The `SampleBernoulli` implementation decomposes the f64 and uses the bits in the mantissa to weight the probability.
Here:
https://github.com/opendp/opendp/blob/9ef96dccd4230552dded0b8995db3ea568ade6af/rust/opendp/src/meas/mod.rs#L150-L159

To make the geometric mechanism generic over the output distance, I need to make this snippet generic with respect to the `Ieee754::Significand` associated type. I haven't found a clean way to do this yet.

Reply:
Cool thanks for the explanation!